protocol/signalfx/trace_jaeger_test.go

```diff
@@ -0,0 +1,479 @@
+              "vType": "STRING",
+              "vStr": "http://127.0.0.1:15598/client_transactions"
+            },
+            {
+              "key": "span.kind",
+              "vType": "STRING",
+              "vStr": "server"
+            },
+            {
+              "key": "peer.port",
+              "vType": "LONG",
+              "vLong": 53931
+            },
+            {
+              "key": "someBool",
+              "vType": "BOOL",
+              "vBool": true
+            },
+            {
+              "key": "someDouble",
+              "vType": "DOUBLE",
+              "vDouble": 129.8
+            },
+            {
+              "key": "peer.service",
+              "vType": "STRING",
+              "vStr": "rtapi"
+            },
+            {
+              "key": "peer.ipv4",
+              "vType": "LONG",
+              "vLong": 3224716605
+            }
+          ],
+          "logs": [
+            {
+              "timestamp": 1485467191639875,
+              "fields": [
+                {
+                  "key": "key1",
+                  "vType": "STRING",
+                  "vStr": "value1"
+                },
+                {
+                  "key": "key2",
+                  "vType": "STRING",
+                  "vStr": "value2"
+                }
+              ]
+            },
+            {
+              "timestamp": 1485467191639875,
+              "fields": [
+                {
+                  "key": "event",
+                  "vType": "STRING",
+                  "vStr": "nothing"
+                }
+              ]
+            }
+          ]
+        }`
+
+	const numSpans = 100
+	reqJSON := `
+    {
+      "process": {
+        "serviceName": "api",
+        "tags": [
+          {
+            "key": "hostname",
+            "vType": "STRING",
+            "vStr": "api246-sjc1"
+          },
+          {
+            "key": "ip",
+            "vType": "STRING",
+            "vStr": "10.53.69.61"
+          },
+          {
+            "key": "jaeger.version",
+            "vType": "STRING",
+            "vStr": "Python-3.1.0"
+          }
+        ]
+      },
+      "spans": [`
+
+	for i := 0; i < numSpans; i++ {
+		reqJSON += spanJSON + ","
+	}
+
+	reqJSON = strings.TrimSuffix(reqJSON, ",") + `]}`
+
+	var batch jThrift.Batch
+	err := json.Unmarshal([]byte(reqJSON), &batch)
+	if err != nil {
+		panic("test json invalid")
+	}
+	batchBytes, err := thrift.NewTSerializer().Write(&batch)
+	if err != nil {
+		panic("couldn't serialize test batch to thrift")
+	}
+
+	reader := bytes.NewReader(batchBytes)
+	reqBody := ioutil.NopCloser(reader)
+
+	decoder := NewJaegerThriftTraceDecoderV1(nil,
+		&fakeSink{
+			handler: func(_ []*trace.Span) {},
+		})
+
+	req := http.Request{
+		Body: reqBody,
+	}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		err := decoder.Read(context.Background(), &req)
+		if err != nil {
+			fmt.Println(i)
+			b.Fatal(err)
+		}
+		reader.Reset(batchBytes)
```

Comment:
b.StopTimer() and b.StartTimer() around the reader.Reset() call would be more a more accurate benchmark.

Reply:
For some strange reason when I do that it makes the ns/op measure go consistently up a few %:
```
BenchmarkJaegerTraceDecoder-8              10000           1118437 ns/op          553094 B/op      10827 allocs/op
```

but I changed it here.

Reply:
weird. would have been a good opporunity to use https://godoc.org/golang.org/x/tools/cmd/benchcmp